src/events/messageCreate/train.ts

```diff
@@ -0,0 +1,34 @@
+import { ArgsOf, Discord, On } from "discordx";
+
+let activeTrain = false;
+let previousString = "";
+let previousUser = "";
+@Discord()
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+abstract class Train {
+	@On({ event: "messageCreate" })
+	private async handleTrain([msg]: ArgsOf<"messageCreate">) {
+		if (!activeTrain) {
+			// eslint-disable-next-line no-void
+			void msg.channel.messages.fetch({ limit: 2 }).then(async (msgs) => {
+				for await (const element of msgs) {
```

Comment:
See comment for line #31. Not sure about the edge cases for this but we can potentially not keep track of state ourselves and compare the msg content/user of the elements in this array instead